src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestIdentitiesDataStore.java

```diff
@@ -0,0 +1,129 @@
+import com.google.common.collect.HashBiMap;
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.colony.requestsystem.data.IRequestIdentitiesDataStore;
+import com.minecolonies.api.colony.requestsystem.factory.FactoryVoidInput;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.TypeConstants;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_LIST;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_REQUEST;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_TOKEN;
+
+/**
+ * A default implementation of the {@link IRequestIdentitiesDataStore} interface.
+ */
+public class StandardRequestIdentitiesDataStore implements IRequestIdentitiesDataStore
+{
+    private IToken<?>                     id;
+    private final BiMap<IToken<?>, IRequest<?>> map;
+
+    public StandardRequestIdentitiesDataStore(
+      final IToken<?> id,
+      final BiMap<IToken<?>, IRequest<?>> map) {
+        this.id = id;
+        this.map = map;
+    }
+
+    public StandardRequestIdentitiesDataStore()
+    {
+        this.id = StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN);
+        this.map = HashBiMap.create();
+    }
+
+    @Override
+    public BiMap<IToken<?>, IRequest<?>> getIdentities()
+    {
+        return map;
+    }
+
+    @Override
+    public IToken<?> getId()
+    {
+        return id;
+    }
+
+    @Override
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestIdentitiesDataStore>
+    {
+
+        @NotNull
+        @Override
+        public TypeToken<? extends StandardRequestIdentitiesDataStore> getFactoryOutputType()
+        {
+            return TypeToken.of(StandardRequestIdentitiesDataStore.class);
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<? extends FactoryVoidInput> getFactoryInputType()
+        {
+            return TypeConstants.FACTORYVOIDINPUT;
+        }
+
+        @NotNull
+        @Override
+        public StandardRequestIdentitiesDataStore getNewInstance(
+          @NotNull final IFactoryController factoryController, @NotNull final FactoryVoidInput factoryVoidInput, @NotNull final Object... context) throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)